src/main/scala/io/scalac/tezos/translator/model/LibraryEntry.scala

```diff
@@ -8,7 +8,7 @@ case class LibraryEntry(
   uid: Uid,
   name: String,
   author: String,
-  email: Option[String],
+  email: Option[EmailAddress],
```

Comment:
Isn't a better idea to use option in EmailAdress if it only contain one String ?

Reply:
do you mean to model `EmailAddress` as a `EmailAddress(v: Option[String])` ?

Reply:
yes

Reply:
in this option you will always have serialized case class EmailAddress even if it contains null

Reply:
No its not a better idea. Solely purpose of `EmailAddress` is to represent a correct email address, so when I stumble across an instance of it in any place in the code I can know it contains a valid email address. Also it is more convenient from compositionality side.
This optionality you are talking about could and imo should be handled outside of such a model. 

Having an option inside would mean that a more defence like programming style would have to be used. This refers to passing `Option` of smth in general.
Consider such situation:
```
def maybeDoSmth(x: Option[Int]): ???
```
vs
```
def doSmth(x: Int): ???
```
I would argue that optionality in `maybeDoSmth` can be handled outside of this method, on the caller side. There are tools like `.map`, `.flatMap`, `for` etc to nicely compose such a monadic wrappings
It is easier to think of function/a piece of code when an argument is given in a straightforward manner. Which method you would prefer to test ?

Going back to `EmailAddress` case, imagine you get a `EmailAddress` instance with `Option` inside, so each time you have to check if its defined or not to perform an action, while on the other side you may not perform an action in the first place if its `None`. Would it make sense to have:
```
def sendEmail(address: EmailAddress): Future[Unit] = address.v match  {
  case Some(v) => actuallySend(address)
  case None => ??? // Future.failed ? Future.success ?
}
```
So in terms of `None` shoudl it be success or failure - IMO we shouldnt get to this point, so no problem like that should occur.

Please let me know it this is clear what im saying. Im happy to discuss more on that

Reply:
okay, i understand. Thank you for sharing knowledge it's very helpful, valuable and teaching.